app/bluetoothdiscoverymodel.cpp

```diff
@@ -0,0 +1,167 @@
+
+  connect( mDiscoveryAgent.get(), QOverload<QBluetoothDeviceDiscoveryAgent::Error>::of( &QBluetoothDeviceDiscoveryAgent::error ),
+           [ = ]( QBluetoothDeviceDiscoveryAgent::Error error )
+  {
+    qDebug() << error << "occured during discovery, ending..";
+    CoreUtils::log( "Bluetooth discovery", QString( "Error occured during device discovery, error code #" ).arg( error ) );
+    finishedDiscovery();
+  } );
+}
+
+BluetoothDiscoveryModel::~BluetoothDiscoveryModel()
+{
+}
+
+QHash<int, QByteArray> BluetoothDiscoveryModel::roleNames() const
+{
+  QHash<int, QByteArray> roles;
+  roles.insert( DataRoles::DeviceAddress, "DeviceAddress" );
+  roles.insert( DataRoles::DeviceName, "DeviceName" );
+  roles.insert( DataRoles::SignalStrength, "SignalStrength" );
+
+  return roles;
+}
+
+int BluetoothDiscoveryModel::rowCount( const QModelIndex & ) const
+{
+  return mFoundDevices.count();
+}
+
+QVariant BluetoothDiscoveryModel::data( const QModelIndex &index, int role ) const
+{
+  if ( !index.isValid() )
+    return QVariant();
+
+  int deviceIndex = index.row();
+
+  if ( deviceIndex < 0 || deviceIndex >= mFoundDevices.count() )
+    return QVariant();
+
+  QBluetoothDeviceInfo device = mFoundDevices[deviceIndex];
+
+  switch ( role )
+  {
+    case DataRoles::DeviceAddress:
+    {
+      return device.address().toString();
+    }
+
+    case DataRoles::DeviceName:
+    {
+      return device.name();
+    }
+
+    case DataRoles::SignalStrength:
+    {
+      return device.rssi();
+    }
+
+    default: return QVariant();
+  }
+}
+
+bool BluetoothDiscoveryModel::discovering() const
+{
+  return mDiscovering;
+}
+
+void BluetoothDiscoveryModel::setDiscovering( bool discovering )
+{
+  if ( mDiscovering == discovering )
+    return;
+
+  if ( discovering )
+  {
+    mDiscoveryAgent->start();
+    CoreUtils::log( QStringLiteral( "Bluetooth discovery" ), QStringLiteral( "Started discovering devices, method %1" ).arg( mDiscoveryAgent->supportedDiscoveryMethods() ) );
+  }
+  else
+  {
+    mDiscoveryAgent->stop();
+  }
+
+  mDiscovering = discovering;
+  emit discoveringChanged( mDiscovering );
+}
+
+// helper class to print the device, let's keep it there for a while
+void _printDeviceInfo( const QBluetoothDeviceInfo &device )
+{
+  qDebug() << "Device" << device.name() << "Address:" << device.address();
+  qDebug() << "  uuid:" << device.deviceUuid();
+  qDebug() << "  rssi (signal strength):" << device.rssi();
+  qDebug() << "  core config:" << device.coreConfigurations();
+  qDebug() << "  service classes:" << device.serviceClasses();
+  qDebug() << "  service uuids:" << device.serviceUuids();
+  qDebug() << "  valid:" << device.isValid();
+  qDebug() << "  cached: " << device.isCached();
+  qDebug() << "  major class:" << device.majorDeviceClass();
+  qDebug() << "  minor class:" << device.minorDeviceClass();
+  qDebug() << " --- ";
```

Comment:
canot this go to the Input App diagnostics log instead?

Reply:
I will remove it in the next PR, we do not need all these information, it is there just for debug purposes.